org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/LocalGradleDistribution.java

```diff
@@ -32,11 +31,23 @@
     private final File location;
 
     LocalGradleDistribution(File location) {
-        this.location = Preconditions.checkNotNull(location);
-        Optional<String> error = GradleDistributionInfo.validateLocalInstallationLocation(location);
+        this.location = Preconditions.checkNotNull(location).getAbsoluteFile();
+        Optional<String> error = validate(location);
         Preconditions.checkArgument(!error.isPresent(), error.or(""));
     }
 
+    private static  Optional<String> validate(File location) {
+        if (location == null) {
+            return Optional.of(NLS.bind(CoreMessages.ErrorMessage_0_MustBeSpecified, CoreMessages.GradleDistribution_Label_LocalInstallationDirectory));
+        } else if (!location.exists()) {
```

Comment:
❌ We shouldn't be doing IO here. This is a value object and shouldn't care about whether this file actually exists when it is constructed. This fact can change over time.

Reply:
Removed.